### PR TITLE
Notification services

### PR DESCRIPTION
**Pull request by HarryFromSpace**
This pull request contains following functionality:
1. Installed RabbitMQ in docker container.  
2. Nofitication framework spring boot application with RabbitMQ listener which is subscribing RabbitMQ data queue.

Tested as follows:
1.

### CODE REVIEW

File: microservice-ecms-notification/src/main/java/com/ecms/MicroserviceEcmsNotificationApplication.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.ecms;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
+
+@EnableSwagger2
+@SpringBootApplication
+// @EnableDiscoveryClient
```
Comment (tanushreeyadav) [inline microservice-ecms-notification/src/main/java/com/ecms/MicroserviceEcmsNotificationApplication.java]: Why committing commented code? please remove it.

File: microservice-ecms-notification/src/main/java/com/ecms/configuration/RabbitMqConfig.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.springframework.amqp.rabbit.config.SimpleRabbitListenerContainerFactory;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter;
+import org.springframework.amqp.support.converter.MessageConverter;
+import org.springframework.boot.autoconfigure.amqp.SimpleRabbitListenerContainerFactoryConfigurer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
```
Comment (tanushreeyadav) [inline microservice-ecms-notification/src/main/java/com/ecms/configuration/RabbitMqConfig.java]: Please define class level comments.

**Review by tanushreeyadav — CHANGES_REQUESTED** [review]
Please make the changes as added in comments.

File: microservice-ecms-notification/src/main/java/com/ecms/service/NotificationService.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+			String html = templateEngine.process("email-template", context);
+			helper.setTo(mail.getTo());
+			helper.setText(html, true);
+			helper.setSubject(mail.getSubject());
+			helper.setFrom(mail.getFrom());
+			javamailSender.send(message);
+			response = "Mail Has Been Successfully Send to : " + mail.getTo();
+		} catch (MessagingException e) {
+			response = "There Seems to be an error in Sending Mail, Please Check";
+			e.printStackTrace();
```
Comment (bigdatapoc) [inline microservice-ecms-notification/src/main/java/com/ecms/service/NotificationService.java]: Need to use log no e.printStackTrace

File: microservice-ecms-notification/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+spring.mail.host=smtp.gmail.com
+spring.mail.username=
+spring.mail.password=
+spring.mail.port=587
+spring.mail.protocol=smtp
+spring.mail.test-connection=false
+spring.mail.properties.mail.smtp.auth=true
+spring.mail.properties.mail.smtp.starttls.enable=true
+
+#RabbitMQ
```
Comment (HarryFromSpace) [inline microservice-ecms-notification/src/main/resources/application.properties]: hi

**Pull request closed without merging**